crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -470,23 +470,39 @@ impl CachedRenderPipelinePhaseItem for Transparent3d {
 
 pub fn extract_core_3d_camera_phases(
     mut commands: Commands,
+    mut opaque_3d_phases: ResMut<ViewBinnedRenderPhases<Opaque3d>>,
+    mut alpha_mask_3d_phases: ResMut<ViewBinnedRenderPhases<AlphaMask3d>>,
+    mut transmissive_3d_phases: ResMut<ViewSortedRenderPhases<Transmissive3d>>,
+    mut transparent_3d_phases: ResMut<ViewSortedRenderPhases<Transparent3d>>,
     cameras_3d: Extract<Query<(Entity, &Camera), With<Camera3d>>>,
 ) {
+    let mut live_entities = EntityHashSet::default();
     for (entity, camera) in &cameras_3d {
-        if camera.is_active {
-            commands.get_or_spawn(entity).insert((
-                BinnedRenderPhase::<Opaque3d>::default(),
-                BinnedRenderPhase::<AlphaMask3d>::default(),
-                SortedRenderPhase::<Transmissive3d>::default(),
-                SortedRenderPhase::<Transparent3d>::default(),
-            ));
+        if !camera.is_active {
+            continue;
         }
+
+        commands.get_or_spawn(entity);
+
+        opaque_3d_phases.insert_or_clear(entity);
+        alpha_mask_3d_phases.insert_or_clear(entity);
+        transmissive_3d_phases.insert_or_clear(entity);
+        transparent_3d_phases.insert_or_clear(entity);
+
+        live_entities.insert(entity);
     }
+
+    opaque_3d_phases.retain(|entity, _| live_entities.contains(entity));
+    alpha_mask_3d_phases.retain(|entity, _| live_entities.contains(entity));
```

Comment:
reading further i see you're already mostly doing the live view retaining. is is intentional to omit transmissive and transparent phases here?

Reply:
Nope, fixed.